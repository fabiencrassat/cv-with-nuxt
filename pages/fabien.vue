<template>
  <main>
    <h1>{{ $t('greeting') }}</h1>

    <nuxt-link v-if="$i18n.locale !== 'en'" :to="switchLocalePath('en')">
      English
    </nuxt-link>

    <nuxt-link v-if="$i18n.locale !== 'fr'" :to="switchLocalePath('fr')">
      Fran√ßais
    </nuxt-link>
  </main>
</template>

<script>
export default {};
</script>
