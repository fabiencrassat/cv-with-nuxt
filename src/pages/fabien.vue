<template>
  <div class="font-display">
    <LeftSideNavigation
      :name="CurriculumVitae.getFullName()"
      :picture="CurriculumVitae.getPicture()"
      :menu-items="[
        {
          name: $t('identity.name'),
          url: 'identity',
          svg: UserSvgIcon,
        },
        {
          name: $t('contact-me.name'),
          url: 'contact-me',
          svg: PhoneSvgIcon,
        },
        {
          name: $t('experiences.name'),
          url: 'experiences',
          svg: BriefcaseSvgIcon,
        },
        {
          name: $t('skills.name'),
          url: 'skills',
          svg: ColoursSvgIcon,
        },
        {
          name: $t('educations.name'),
          url: 'educations',
          svg: GraduationCapSvgIcon,
        },
        {
          name: $t('languages.name'),
          url: 'languages',
          svg: LanguageSvgIcon,
        },
        {
          name: $t('certifications.name'),
          url: 'certifications',
          svg: MedalSvgIcon,
        },
        {
          name: $t('hobbies.name'),
          url: 'hobbies',
          svg: HeartSvgIcon,
        },
      ]"
    />
    <main class="p-4">
      <section id="identity">
        <H1Heading :first="true">
          {{ $t('identity.name') }}
        </H1Heading>
        <p class="text-justify">
          <span class="text-lg text-blue-700 italic whitespace-no-wrap">
            39 years
          </span>
          &nbsp;/&nbsp;
          <span class="text-lg text-blue-700 italic whitespace-no-wrap">
            Paris
          </span>
          &nbsp;/&nbsp;
          <span class="text-lg text-blue-700 italic">
            {{ CurriculumVitae.getLastJob() }}
          </span>
        </p>
        <p class="text-justify">
          {{ CurriculumVitae.getPresentation() }}
        </p>
      </section>
      <section id="contact-me">
        <H1Heading>{{ $t('contact-me.name') }}</H1Heading>
        <FollowMe :links="CurriculumVitae.getFollowMe()" />
        <ContactMe />
      </section>
      <section id="experiences">
        <H1Heading>{{ $t('experiences.name') }}</H1Heading>
        <Experiences :experiences="CurriculumVitae.getExperiences()" />
      </section>
      <section id="skills">
        <H1Heading>{{ $t('skills.name') }}</H1Heading>
        <Skills :skills="CurriculumVitae.getSkills()" />
      </section>
      <section id="educations">
        <H1Heading>{{ $t('educations.name') }}</H1Heading>
        <Educations :educations="CurriculumVitae.getEducations()" />
      </section>
      <section id="languages">
        <H1Heading>{{ $t('languages.name') }}</H1Heading>
        <Languages :languages="CurriculumVitae.getLanguages()" />
      </section>
      <section id="certifications">
        <H1Heading>{{ $t('certifications.name') }}</H1Heading>
        <Certifications :certifications="CurriculumVitae.getCertifications()" />
      </section>
      <section id="hobbies">
        <H1Heading>{{ $t('hobbies.name') }}</H1Heading>
        <Hobbies :hobbies="CurriculumVitae.getHobbies()" />
      </section>
    </main>
  </div>
</template>

<script>
import BriefcaseSvgIcon from '~/components/svgIcons/briefcase';
import Certifications from '~/components/certifications/certifications';
import ColoursSvgIcon from '~/components/svgIcons/colours';
import ContactMe from '~/components/contact-me/contact-me';
import Educations from '~/components/educations/educations';
import Experiences from '~/components/experiences/experiences';
import FollowMe from '~/components/follow-me/followMe';
import GraduationCapSvgIcon from '~/components/svgIcons/graduation-cap';
import H1Heading from '~/components/headings/h1.vue';
import Hobbies from '~/components/hobbies/hobbies.vue';
import Languages from '~/components/languages/languages';
import LanguageSvgIcon from '~/components/svgIcons/language';
import LeftSideNavigation from '~/components/navigations/leftSide.vue';
import HeartSvgIcon from '~/components/svgIcons/heart.vue';
import MedalSvgIcon from '~/components/svgIcons/medal.vue';
import Skills from '~/components/skills/skills';
import PhoneSvgIcon from '~/components/svgIcons/phone';
import UserSvgIcon from '~/components/svgIcons/user';
import CurriculumVitae from '~/lib/curriculumVitae';

export default {
  components: {
    // eslint-disable-next-line vue/no-unused-components
    BriefcaseSvgIcon,
    Certifications,
    // eslint-disable-next-line vue/no-unused-components
    ColoursSvgIcon,
    ContactMe,
    Educations,
    Experiences,
    FollowMe,
    // eslint-disable-next-line vue/no-unused-components
    GraduationCapSvgIcon,
    H1Heading,
    Hobbies,
    Languages,
    // eslint-disable-next-line vue/no-unused-components
    LanguageSvgIcon,
    LeftSideNavigation,
    // eslint-disable-next-line vue/no-unused-components
    HeartSvgIcon,
    // eslint-disable-next-line vue/no-unused-components
    MedalSvgIcon,
    // eslint-disable-next-line vue/no-unused-components
    PhoneSvgIcon,
    Skills,
    // eslint-disable-next-line vue/no-unused-components
    UserSvgIcon,
  },
  data() {
    // eslint-disable-next-line no-use-before-define
    const curriculumVitae = new CurriculumVitae(this.$i18n.locale);

    return {
      BriefcaseSvgIcon,
      ColoursSvgIcon,
      description:
        curriculumVitae.getLastJob() + '. ' + curriculumVitae.getPresentation(),
      CurriculumVitae: curriculumVitae,
      GraduationCapSvgIcon,
      LanguageSvgIcon,
      HeartSvgIcon,
      MedalSvgIcon,
      path: this.$nuxt.$route.path,
      PhoneSvgIcon,
      shortTitle: this.$i18n.t('page.shortTitle', {
        name: curriculumVitae.getFullName(),
      }),
      title: this.$i18n.t('page.title', {
        name: curriculumVitae.getFullName(),
      }),
      UserSvgIcon,
    };
  },
  head() {
    return {
      title: this.title,
      meta: [
        {
          hid: 'description',
          name: 'description',
          content: this.description,
        },
        {
          hid: 'og:url',
          name: 'og:url',
          content: this.path,
        },
        {
          hid: 'og:site_name',
          name: 'og:site_name',
          content: this.title,
        },
        {
          hid: 'og:title',
          name: 'og:title',
          content: this.title,
        },
        {
          hid: 'og:description',
          name: 'og:description',
          content: this.description,
        },
        {
          hid: 'apple-mobile-web-app-title',
          name: 'apple-mobile-web-app-title',
          content: this.shortTitle,
        },
        {
          hid: 'application-name',
          name: 'application-name',
          content: this.shortTitle,
        },
      ],
    };
  },
};
</script>
