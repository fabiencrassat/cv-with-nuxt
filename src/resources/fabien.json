{
  "identity": {
    "myself": {
      "fullName": "Fabien Crassat",
      "picture": "/fabien-crassat.jpeg"
    }
  },
  "followMe": [
    {
      "label": {
        "en": "Follow me in LinkedIn",
        "fr": "Suivez moi sur LinkedIn"
      },
      "url": "http://www.linkedin.com/in/fabiencrassat",
      "svg": "LinkedInSvgIcon"
    },
    {
      "label": {
        "en": "Follow me in Viadeo",
        "fr": "Suivez moi sur Viadeo"
      },
      "url": "http://fr.viadeo.com/fr/profile/fabien.crassat",
      "svg": "ViadeoSvgIcon"
    },
    {
      "label": {
        "en": "Follow me in Monster",
        "fr": "Suivez moi sur Monster"
      },
      "url": "http://beknown.com/fabien-crassat",
      "svg": "MonsterInSvgIcon"
    },
    {
      "label": {
        "en": "Follow me in Twitter",
        "fr": "Suivez moi sur Twitter"
      },
      "url": "https://twitter.com/fabiencrassat",
      "svg": "TwitterSvgIcon"
    },
    {
      "label": {
        "en": "Follow me in Facebook",
        "fr": "Suivez moi sur Facebook"
      },
      "url": "https://www.facebook.com/fabien.crassat",
      "svg": "FacebookSvgIcon"
    },
    {
      "label": {
        "en": "Follow me in Scrum Alliance",
        "fr": "Suivez moi sur Scrum Alliance"
      },
      "url": "http://www.scrumalliance.org/community/profile/fcrassat",
      "svg": "ScrumAllianceSvgIcon"
    },
    {
      "label": {
        "en": "Follow me in Github",
        "fr": "Suivez moi sur Github"
      },
      "url": "https://github.com/fabiencrassat",
      "svg": "GithubSvgIcon"
    }
  ],
  "lookingFor": {
    "experience": {
      "crossref": "curriculumVitae.experiences.Reacteev.job"
    },
    "presentation": {
      "fr": "10 ans dans une startup grandissante (acquis maintenant) de 30 à 300 personnes utilisant Agile et Lean Startup. Maintenant coach Agile et Lean Startup dédié à la découverte des clients, l'innovation et l'efficacité de l'entreprise.",
      "en": "10+ years scaling a startup (acquired now) from 30 to 300 people using Agile and Lean Startup. Now Agile and Lean Startup coach dedicated to customer discovery, innovation and corporate efficiency."
    }
  },
  "experiences": {
    "Reacteev": {
      "date": {
        "fr": "Septembre 2017 - Aujourd'hui",
        "en": "September 2017 - Present"
      },
      "job": {
        "fr": "Coach Agile, DevOps et Lean Startup",
        "en": "Agile, DevOps and Lean Startup Coach"
      },
      "jobImage": "/img/experiences/coach.webp",
      "society": {
        "crossref": "societies.reacteev"
      },
      "missions": {
        "fr": [
          "Améliore l'efficience des départements informatiques des clients : études, analyses et accompagnements des équipes et partenaires.",
          "Démarche de nouveaux clients à travers la participation à des salons ou événements et de la prospection commerciale."
        ],
        "en": [
          "Improves the efficiency of the IT departments of the clients: studies, analyzes and accompaniments of teams and partners.",
          "Approach new clients through participation in trade shows or events and prospecting."
        ]
      }
    },
    "EnablonInfraSquad": {
      "date": {
        "fr": "Juillet 2015 - Août 2017",
        "en": "July 2015 - August 2017"
      },
      "job": {
        "fr": "Tech Leader d'une équipe agile",
        "en": "Tech Leader in Agile team"
      },
      "jobImage": "/img/experiences/tech-leader.webp",
      "society": {
        "crossref": "societies.enablon"
      },
      "missions": {
        "fr": [
          "Dirige l'équipe de développement du framework d'Enablon et du gestionnaire de source.",
          "Fournie des solutions techniques pertinentes fondées sur les meilleures pratiques / expériences / projets similaires.",
          "Établit la vision technique / architecture avec l'équipe de développement.",
          "Fournie l'examen technique et les estimations des composantes de développement / temps requis pour les projets.",
          "Identifier les risques, les résultats, les voies d'amélioration de l'effort de développement."
        ],
        "en": [
          "Responsible for leading the development team of Enablon framework and source management.",
          "Deliver relevant technical solutions based on best practice / experiences / similar projects.",
          "Establishes technical vision / architecture with the development team.",
          "Provide technical review and estimations of the development components / time required for the projects.",
          "Identify risks, findings and ways of improvement for development effort."
        ]
      }
    },
    "EnablonBackbone": {
      "date": {
        "fr": "Mai 2014 - Juin 2015",
        "en": "May 2014 - June 2015"
      },
      "job": {
        "fr": "Responsable Coordinateur Support Niveau 3",
        "en": "Support Level 3 Lead Coordinator"
      },
      "jobImage": "/img/experiences/support.webp",
      "society": {
        "crossref": "societies.enablon"
      },
      "missions": {
        "fr": [
          "Coordonne toute l'activité du support niveau 3 d'une équipe agile internationale.",
          "Construit et met à jour les tableaux de bord pour le suivi interne et la communication externe.",
          "Unifie le planning prévisionnel des livrables et les Releases Notes.",
          "Gère la relation et assiste aux réunions de suivi des parties prenante en conduisant des questionnaires réguliers pour mesurer le niveau de satisfaction.",
          "Améliore la qualité.",
          "En plus en tant que Architecte logiciel Suite Enablon: Définie et développe un gestionnaire de source (semblable à GIT) incluant l'installation, le commit, la mise à jour et la suppression à travers le cycle de vie du produit."
        ],
        "en": [
          "Coordinate Support level 3 Activity of an international agile team.",
          "Build and update Dashboards (for internal follow-Up and external communication).",
          "Unified Forecasted planning and Releases Notes.",
          "Manage relationship and attend following stakeholders meetings conducting regular surveys to measure the level of satisfaction.",
          "Improve quality.",
          "Extra as Enablon Software Architect: Design and develop a Delivery (Git Like) Tool including \"Installation, Commit, Update, Delete\" across the entire product lifecycle."
        ]
      }
    },
    "EnablonArchitect": {
      "date": {
        "fr": "Avril 2011 - Mai 2014",
        "en": "April 2011 - May 2014"
      },
      "job": {
        "fr": "Architecte logiciel Suite Enablon",
        "en": "Enablon Suite Software Architect"
      },
      "jobImage": "/img/experiences/architect.webp",
      "society": {
        "crossref": "societies.enablon"
      },
      "missions": {
        "fr": [
          "Passionné par l'optimisation de l'architecture, je joue un rôle clé dans l'expansion des solutions Enablon développé par des équipes distribuées répartis dans le monde. Je contribue dans la conception, les coûts et je guide l'équipe pour fournir des logiciels avec un niveau élevé de qualité. En étroite collaboration avec les chefs de produits, les testeurs et les autres ingénieurs logiciels, je conçois, test, code, fournie et maintiens des modules logiciels en utilisant le langage propriétaire Enablon et les langages du web dans un environnement Agile Scrum.",
          "Analyse et guide l'équipe pour résoudre des problèmes complexes.",
          "Travail en étroite collaboration avec l'équipe d'architectes des différentes applications pour discuter des challenges et des solutions pour assurer une consistance à travers la suite Enablon.",
          "Réfléchie et résous de très intéressants problèmes à l'attention directe d'une expérience utilisateur.",
          "Accompagne et guide les membres juniors de l'équipe.",
          "Partage les connaissances à travers la collaboration, les formations, les revues, les présentations, les articles, ...",
          "Responsable des livraisons dans les délais impartis des solutions Enablon.",
          "Participe au recrutement.",
          "Développe un gestionnaire de source (semblable à GIT)."
        ],
        "en": [
          "Passionate about architecture optimization, I am a key contributor in the expansion of the Enablon solution developed by distributed teams located across the world. I am the go-to-person in designing, costing and guiding the team to deliver software within a high quality level. Collaborating closely with product managers, testers and other software engineers, I design, test, code, deliver and maintain software modules using proprietary and non-proprietary language within an Agile Scrum environment.",
          "Troubleshoot and guide the team to resolve complex problems.",
          "Work with team of architects from various applications to discuss challenges and solutions to ensure consistency across the application suite.",
          "Brainstorm and solve very interesting problems that will considerably increase product user experience.",
          "Coach and mentor junior team members.",
          "Share knowledge through collaboration, training, peer review, presentations, articles, etc.",
          "Responsible for delivering in time new application versions to the application suite.",
          "Participate in recruitment.",
          "Development of version control (GIT like)."
        ]
      }
    },
    "EnablonGRCArchitect": {
      "date": {
        "fr": "Novembre 2009 - Avril 2011",
        "en": "November 2009 - April 2011"
      },
      "job": {
        "fr": "Architecte logiciel Suite GRC",
        "en": "GRC Suite Software Architect"
      },
      "jobImage": "/img/experiences/architect2.webp",
      "society": {
        "crossref": "societies.enablon"
      },
      "missions": {
        "fr": [
          "Définition et conception des architectures d’applications complexes des solutions de Gouvernance des Risques et de la Conformité.",
          "Participation au chiffrage détaillé des propositions techniques par la faisabilité technique et les choix technologiques.",
          "Rédaction des propositions techniques et des dossiers d’architecture logicielle.",
          "Définition des prototypes et collaboration aux missions d’avant-vente.",
          "Pérennisation de l'information, valorisation de l'expérience et capitalisation des meilleures pratiques par des documents de déploiement et de transfert de compétences.",
          "Assistance à la livraison des solutions et la mise en place de nouvelles fonctionnalités.",
          "Expertise et accompagnement des clients en avant-vente et durant leur projet."
        ],
        "en": [
          "Architecture definition and design of complex Governance Risk Conformity applications.",
          "Participation in detailed proposals by the technical feasibility and technology choices.",
          "Writing of technical proposals and software architecture cases.",
          "Definition of prototypes and cooperation of pre-sales missions.",
          "Continuity of information, valorisation of experience and capitalizing on best practices with documents of deployment and transfer of skills.",
          "Support the delivery of solutions and the implementation of new features.",
          "Expertise and support to customers in pre-sales then during their project."
        ]
      }
    },
    "EnablonAnalyst": {
      "collapse": true,
      "date": {
        "fr": "Janvier 2007 - Octobre 2009",
        "en": "January 2007 - October 2009"
      },
      "job": {
        "fr": "Analyste Logiciel",
        "en": "Software Analyst"
      },
      "jobImage": "/img/experiences/analyst.webp",
      "society": {
        "crossref": "societies.enablon"
      },
      "missions": {
        "fr": [
          "Définition et réalisation des solutions standards métiers de gestion des risques et de contrôles.",
          "Collaboration aux missions d’avant-vente et réalisation de prototypes.",
          "Support du département conseil dans la livraison des solutions et la mise en place de nouvelles fonctionnalités."
        ],
        "en": [
          "Definition and realisation of risks and controls management suite software.",
          "Definition of prototypes and cooperation of pre-sales missions.",
          "Support the delivery of solutions and the implementation of new features."
        ]
      }
    },
    "EnablonInternship": {
      "collapse": true,
      "date": {
        "fr": "Mars - Décembre 2006",
        "en": "March - December 2006"
      },
      "job": {
        "fr": "Stage de fin d’étude de l’ESIEA",
        "en": "Internship for the end of ESIEA"
      },
      "jobImage": "/img/experiences/developer.webp",
      "society": {
        "crossref": "societies.enablon"
      },
      "missions": {
        "fr": [
          "Analyste et développeur logiciel ; Soutien Avant-vente de logiciel de reporting, spécialement le contrôle interne."
        ],
        "en": [
          "Analyst and software developer; pre-sales support on reporting software, especially internal control."
        ]
      }
    },
    "BostonTeaParty": {
      "collapse": true,
      "date": {
        "fr": "Eté 2005",
        "en": "Summer 2005"
      },
      "job": {
        "fr": "Travail saisonnier : cuisinier, serveur et plongeur",
        "en": "Learnt English language through jobs: cook, waiter and washer"
      },
      "jobImage": "/img/experiences/waiter.webp",
      "society": {
        "crossref": "societies.bostonTeaParty"
      },
      "missions": {
        "fr": [
          "Mission à l’étranger de 4ième année de l’ESIEA."
        ],
        "en": [
          "Traineeship abroad in 4th year of ESIEA."
        ]
      }
    },
    "PAIR": {
      "collapse": true,
      "date": {
        "fr": "Février - Mai 2005",
        "en": "February - May 2005"
      },
      "job": {
        "fr": "Projet à application industrielle de 4ième année de l’ESIEA",
        "en": "‘Projet d’Application Industrielle ou de Recherche’ (PAIR) in 4th year of ESIEA"
      },
      "jobImage": "/img/experiences/wifi-vpn-developer.webp",
      "society": {
        "crossref": "societies.suez"
      },
      "missions": {
        "fr": [
          "Réalisation d’un VPN sécurisé sur un réseau Wifi entre 2 sites industriels."
        ],
        "en": [
          "Made a security VPN on a Wi-Fi network between 2 industrial sites."
        ]
      }
    },
    "Korrigan": {
      "collapse": true,
      "date": {
        "fr": "Eté 2004",
        "en": "Summer 2004"
      },
      "job": {
        "fr": "Stage technique de l’ESIEA",
        "en": "ESIEA Technical internship"
      },
      "jobImage": "/img/experiences/vba-developer.webp",
      "society": {
        "crossref": "societies.korrigan"
      },
      "missions": {
        "fr": [
          "Developpement de macros VBA pour automatiser des extractions en lien avec un outil de gestion."
        ],
        "en": [
          "Development of VBA macros to automate extractions in connection with a management tool."
        ]
      }
    },
    "OBIWarehouseman": {
      "collapse": true,
      "date": {
        "fr": "Eté 2003",
        "en": "Summer 2003"
      },
      "job": {
        "fr": "Manutentionnaire et vendeur dans les rayonnages",
        "en": "Warehouseman on sale in the shelf"
      },
      "jobImage": "/img/experiences/warehouseman.webp",
      "society": {
        "crossref": "societies.obi"
      }
    },
    "SalesEngineer": {
      "collapse": true,
      "date": {
        "fr": "Septembre 2002",
        "en": "September 2002"
      },
      "job": {
        "fr": "Technico-commercial spécialisé dans le traitement de l’eau",
        "en": "Sales Engineer specialized in the water treatment"
      },
      "jobImage": "/img/experiences/water-seller.webp",
      "society": {
        "crossref": "societies.cge"
      },
      "missions": {
        "fr": [
          "Définir les contacts utiles et présentation du produit chez les particuliers, 1 vente réussie."
        ],
        "en": [
          "Worked with the sale department, defined the useful contacts to have appointments and presentation of the product: one successful sale."
        ]
      }
    },
    "DUTInternship": {
      "collapse": true,
      "date": {
        "fr": "Avril - Juin 2002",
        "en": "April - June 2002"
      },
      "job": {
        "fr": "Stage de fin d’étude de l'IUT",
        "en": "Internship for the end of DUT"
      },
      "jobImage": "/img/experiences/technician.webp",
      "society": {
        "crossref": "societies.leseth"
      },
      "missions": {
        "fr": [
          "Conception, réalisation et étalonnage d’un débitmètre de type Venturi en régime transitoire."
        ],
        "en": [
          "Designed, realised and calibrated a flowmeter in transient state."
        ]
      }
    },
    "SVGSOwarehouseman": {
      "collapse": true,
      "date": {
        "fr": "Eté 2001",
        "en": "Summer 2001"
      },
      "job": {
        "fr": "Intérimaire",
        "en": "Temporary worker"
      },
      "jobImage": "/img/experiences/wine-worker.webp",
      "society": {
        "crossref": "societies.svgso"
      },
      "missions": {
        "fr": [
          "Travail temporaire en conditionnement de chaîne d’embouteillage."
        ],
        "en": [
          "Warehouseman in conditioning of chain."
        ]
      }
    }
  },
  "skills": {
    "working": {
      "label": {
        "fr": "Méthode de travail",
        "en": "Way of working"
      },
      "svg": "BriefcaseSvgIcon",
      "items": [
        {
          "percentage": 90,
          "label": "Agile, SCRUM, Kanban, DevOps, Craftsmanship"
        },
        {
          "percentage": 60,
          "label": "Lean Startup, Design Thinking"
        }
      ]
    },
    "functional" : {
      "label": {
        "fr": "Fonctionnel",
        "en": "Fonctionnal"
      },
      "svg": "ColoursSvgIcon",
      "items": [
        {
          "percentage": 90,
          "label": {
            "fr": "Gouvernance des Risques, Contrôle Interne et Permanent, Audit, Plans de continuité d'activités",
            "en": "Risk Governance, Internal and Continuous Control, Audit, Business Continuity Management"
          }
        },
        {
          "percentage": 80,
          "label": {
            "fr": "Plan d'action, Circuit d'approbation",
            "en": "Action Plan, Approval Process"
          }
        },
        {
          "percentage": 40,
          "label": {
            "fr": "Management des incidents, Délégation des autorisations et des signatures",
            "en": "Event Management, Authorized and Sign Management"
          }
        }
      ]
    },
    "languages" : {
      "label": {
        "fr": "Langages",
        "en": "Languages"
      },
      "svg": "CodeSvgIcon",
      "items": [
        {
          "percentage": 90,
          "label": "Enablon"
        },
        {
          "percentage": 80,
          "label": "HTML, CSS, JavaScript, jQuery"
        },
        {
          "percentage": 40,
          "label": "C, Data Base (Access, PL/SQL, Oracle), Java, Visual Basic, Assembler, TeX, Media Wiki, Symfony2, NodeJS, Meteor"
        },
        {
          "percentage": 40,
          "label": "Network, Prolog, Interpreter (Lex, Yacc)"
        }
      ]
    },
    "softwares" : {
      "label": {
        "fr": "Logiciels",
        "en": "Softwares"
      },
      "svg": "BrowserSvgIcon",
      "items": [
        {
          "percentage": 80,
          "label": "Word, Excel, PowerPoint, MS Project"
        },
        {
          "percentage": 60,
          "label": "GIT"
        },
        {
          "percentage": 40,
          "label": "Photoshop CS"
        },
        {
          "percentage": 20,
          "label": "Maple, Matlab, LabView, LabCVI, Aphelion, AutoCAD, Quark, Illustrator CS, Pspice"
        }
      ]
    },
    "environments" : {
      "label": {
        "fr": "Environnements",
        "en": "Environments"
      },
      "svg": "LaptopSvgIcon",
      "items": [
        {
          "percentage": 80,
          "label": "Microsoft Windows"
        },
        {
          "percentage": 40,
          "label": "Linux"
        }
      ]
    }
  }
}
