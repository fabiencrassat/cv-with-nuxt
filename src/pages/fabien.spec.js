import testUtils from '~~/lib/tests/testUtils';
import fabien from './fabien';

jest.mock('~/components/navigations/leftSide.vue', () => jest.fn);

const stubs = ['nuxt-link', 'LeftSideNavigation'];

describe('fabien', () => {
  it('mounts properly', () => {
    expect.hasAssertions();
    testUtils.mountProperly(fabien, { stubs });
  });

  it('renders properly', () => {
    expect.hasAssertions();
    // eslint-disable-next-line jest/prefer-inline-snapshots
    expect(testUtils.htmlFactory(fabien, { stubs })).toMatchSnapshot();
  });
  it('renders properly en lang', () => {
    expect.hasAssertions();
    expect(
      testUtils.htmlFactory(fabien, { stubs }, { locale: 'en' })
      // eslint-disable-next-line jest/prefer-inline-snapshots
    ).toMatchSnapshot();
  });
  it('renders properly fr lang', () => {
    expect.hasAssertions();
    expect(
      testUtils.htmlFactory(fabien, { stubs }, { locale: 'fr' })
      // eslint-disable-next-line jest/prefer-inline-snapshots
    ).toMatchSnapshot();
  });

  it('return head', () => {
    expect.hasAssertions();
    expect(testUtils.metaHeadInfoFactory(fabien, { stubs })).toStrictEqual({
      meta: [
        {
          content: "this.$i18n.t('page.description')",
          hid: 'description',
          name: 'description',
        },
        { content: 'this.$nuxt.$route.path', hid: 'og:url', name: 'og:url' },
        {
          content: "this.$i18n.t('page.title')",
          hid: 'og:site_name',
          name: 'og:site_name',
        },
        {
          content: "this.$i18n.t('page.title')",
          hid: 'og:title',
          name: 'og:title',
        },
        {
          content: "this.$i18n.t('page.description')",
          hid: 'og:description',
          name: 'og:description',
        },
      ],
      title: "this.$i18n.t('page.title')",
    });
  });
});
