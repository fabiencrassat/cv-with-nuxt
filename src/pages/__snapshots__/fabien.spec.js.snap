// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fabien renders properly 1`] = `
<div>
  <leftsidenavigation-stub name="getFullName" picture="getPicture" menuitems="[object Object],[object Object],[object Object],[object Object]"></leftsidenavigation-stub>
  <main class="relative h-screen p-4 pt-10 sm:pt-4">
    <section id="$t('identity.url')">
      <h1heading-stub>$t('identity.name')</h1heading-stub>
      <p class="text-justify"><span class="text-lg text-blue-700 italic whitespace-no-wrap">
          39 years
        </span>
        &nbsp;/&nbsp;
        <span class="text-lg text-blue-700 italic whitespace-no-wrap">
          Paris
        </span>
        &nbsp;/&nbsp;
        <span class="text-lg text-blue-700 italic">
          getLastJob
        </span></p>
      <p class="text-justify">
        getPresentation
      </p> <a target="_blank" aria-label="Follow me in LinkedIn" href="http://www.linkedin.com/in/fabiencrassat" rel="noopener noreferrer">
        <linkedinsvgicon-stub></linkedinsvgicon-stub>
      </a> <a target="_blank" aria-label="Follow me in Viadeo" href="http://fr.viadeo.com/fr/profile/fabien.crassat" rel="noopener noreferrer">
        <viadeosvgicon-stub></viadeosvgicon-stub>
      </a> <a target="_blank" aria-label="Follow me in Monster" href="http://beknown.com/fabien-crassat" rel="noopener noreferrer">
        <monsterinsvgicon-stub></monsterinsvgicon-stub>
      </a> <a target="_blank" aria-label="Follow me in Twitter" href="https://twitter.com/fabiencrassat" rel="noopener noreferrer">
        <twittersvgicon-stub></twittersvgicon-stub>
      </a> <a target="_blank" aria-label="Follow me in Facebook" href="https://www.facebook.com/fabien.crassat" rel="noopener noreferrer">
        <facebooksvgicon-stub></facebooksvgicon-stub>
      </a> <a target="_blank" aria-label="Follow me in Scrum Alliance" href="http://www.scrumalliance.org/community/profile/fcrassat" rel="noopener noreferrer">
        <scrumalliancesvgicon-stub></scrumalliancesvgicon-stub>
      </a> <a target="_blank" aria-label="Follow me in Github" href="https://github.com/fabiencrassat" rel="noopener noreferrer">
        <githubsvgicon-stub></githubsvgicon-stub>
      </a>
    </section>
    <section id="$t('contactMe.url')">
      <h1heading-stub>$t('contactMe.name')</h1heading-stub>
      <p><svg id="Location_pin" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" x="0px" y="0px" viewBox="0 0 20 20" enable-background="new 0 0 20 20" xml:space="preserve" class="h-5 w-5 inline fill-current text-blue-700">
          <path d="M10,2.009c-2.762,0-5,2.229-5,4.99c0,4.774,5,11,5,11s5-6.227,5-11C15,4.239,12.762,2.009,10,2.009z M10,9.76c-1.492,0-2.7-1.209-2.7-2.7s1.208-2.7,2.7-2.7c1.49,0,2.699,1.209,2.699,2.7S11.49,9.76,10,9.76z"></path>
        </svg> <a href="http://goo.gl/maps/UJrFf" target="_blank" rel="noopener noreferrer">
          Paris, France
        </a></p>
    </section>
    <section id="$t('experiences.url')">
      <h1heading-stub>$t('experiences.name')</h1heading-stub>
      <ul class="flex flex-wrap">
        <li class="flex-grow my-2 sm:m-4 rounded-sm border-2 border-solid border-blue-700">
          <h2><span>job1</span> <span class="whitespace-no-wrap float-right">
              date1
            </span></h2>
          <ul>
            <li>
              missionItem1
            </li>
          </ul>
          <div><a href="siteurl" target="_blank" rel="noopener noreferrer">
              society1, address1
            </a></div>
        </li>
        <li class="flex-grow my-2 sm:m-4 rounded-sm border-2 border-solid border-blue-700">
          <h2><span>job2</span> <span class="whitespace-no-wrap float-right">
              date2
            </span></h2>
          <ul></ul>
          <div>
            <p>society2, address2</p>
          </div>
        </li>
        <li class="flex-grow my-2 sm:m-4 rounded-sm border-2 border-solid border-blue-700">
          <h2><span>job3</span> <span class="whitespace-no-wrap float-right">
              date3
            </span></h2>
          <ul></ul>
          <!---->
        </li>
      </ul>
    </section>
    <section id="$t('skill.url')">
      <h1heading-stub>$t('skill.name')</h1heading-stub>
    </section>
  </main>
</div>
`;
